### PR TITLE
 Bug 1426507 Upgrade BMO to HTML5

### PR DESCRIPTION
**Pull request by Fatehsandhu**
@kyoshino Hey, could you let me know if this is good? And anything else you need me to do?
Thanks!

### CODE REVIEW

File: extensions/REMO/template/en/default/bug/create/create-remo-swag.html.tmpl [hunk 1]
```diff
@@ -27,7 +27,7 @@
 <p>These requests will only be visible to the person who submitted the request,
 any persons designated in the CC line, and authorized members of the Mozilla Rep team.</p>
 
-<script language="javascript" type="text/javascript">
+<script language="javascript">
```
Comment (kyoshino) [inline extensions/REMO/template/en/default/bug/create/create-remo-swag.html.tmpl]: Remove ` language="javascript"`

File: extensions/REMO/template/en/default/pages/remo-form-payment.html.tmpl [hunk 2]
```diff
@@ -30,7 +30,7 @@
                        'js/field.js' ]
 %]
 
-<script language="javascript" type="text/javascript"></script>
+<script language="javascript"></script>
```
Comment (kyoshino) [inline extensions/REMO/template/en/default/pages/remo-form-payment.html.tmpl]: This `<script>` is empty so you can simply remove this line.

File: template/en/default/admin/users/edit.html.tmpl [hunk 3]
```diff
@@ -99,7 +99,7 @@ $(function() {
                   [% %]<input type="checkbox"
                               name="bless_[% group.id %]"
                               value="1"
-                              [% ' checked="checked"' IF perms.directbless %] />
+                              [% ' checked="checked"' IF perms.directbless %] >
```
Comment (kyoshino) [inline template/en/default/admin/users/edit.html.tmpl]: Remove space before `>`

File: template/en/default/admin/users/edit.html.tmpl [hunk 4]
```diff
@@ -109,7 +109,7 @@ $(function() {
                            id="group_[% group.id %]"
                            name="group_[% group.id %]"
                            value="1"
-                           [% ' checked="checked"' IF perms.directmember %] />
+                           [% ' checked="checked"' IF perms.directmember %] >
```
Comment (kyoshino) [inline template/en/default/admin/users/edit.html.tmpl]: Remove space before `>`

File: template/en/default/attachment/createformcontents.html.tmpl [hunk 5]
```diff
@@ -58,7 +58,7 @@
     <input type="checkbox" id="ispatch" name="ispatch" value="1">
     <label for="ispatch">patch</label><br><br>
     [%# Reset this whenever the page loads so that the JS state is up to date %]
-    <script type="text/javascript" [% csp_nonce FILTER none %]>
+    <script  [% csp_nonce FILTER none %]>
```
Comment (kyoshino) [inline template/en/default/attachment/createformcontents.html.tmpl]: Double space.

File: template/en/default/reports/report-table.html.tmpl [hunk 6]
```diff
@@ -128,7 +128,7 @@ YAHOO.util.Event.addListener(window, "load", function() {
 
     // Do not use <\/a>$. For some reason, IE6 doesn't understand it.
     // We use [^\d]+$ instead.
-    var res = str.match(/>(\d+)[^\d]+$/m);
+    var res = str.match(>(\d+)[^\d]+$/m);
```
Comment (kyoshino) [inline template/en/default/reports/report-table.html.tmpl]: This slash is required as part of JavaScript regular expression.

File: template/en/default/global/header.html.tmpl [hunk 7]
```diff
@@ -399,6 +399,6 @@
 [% END %]
 
 [% BLOCK format_js_link %]
-  <script [% script_nonce FILTER none %] type="text/javascript" src="[% asset_url FILTER version FILTER html %]"></script>
+  <script [% script_nonce FILTER none %]  src="[% asset_url FILTER version FILTER html %]"></script>
```
Comment (kyoshino) [inline template/en/default/global/header.html.tmpl]: Double space.

**Comment by Fatehsandhu** [comment]
@kyoshino Sorry for all the mess. I added something for charset. Can you please have a look?
thanks!

**Comment by kyoshino** [comment]
Actually the legacy charset exists on line 133 in `header.html.tmpl`, so instead of adding a new line, it's better to replace it with the new one.

File: template/en/default/global/header.html.tmpl [hunk 8]
```diff
@@ -131,7 +130,7 @@
     <title>[% title %]</title>
 
     [% IF Param('utf8') %]
-      <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+      <meta charset="UTF-8">
```
Comment (kyoshino) [inline template/en/default/global/header.html.tmpl]: 👍

Comment (Fatehsandhu) [reply template/en/default/global/header.html.tmpl]: Happy New Years!!!
And anything else?

Comment (kyoshino) [reply template/en/default/global/header.html.tmpl]: I think that's it! Let's just wait for the tests to finish.

**Comment by kyoshino** [comment]
Oh, can I ask you one more thing? Replace `<style type="text/css">` with `<style>` in 6 HTML files (excluding one SVG file.) With HTML5, the `type` attribute is optional like `<script>`.

**Comment by Fatehsandhu** [comment]
@kyoshino Should I just make these changes in the ".html.tmpl" files? And also in the `<link>` tags as well?

**Comment by kyoshino** [comment]
As I quickly checked, 4 `<style type="text/css">` in `.html` and 2 in `.html.tmpl`. `<link>` still requires the type.

**Comment by Fatehsandhu** [comment]
@kyoshino Thanks a lot for your help!

**Comment by Fatehsandhu** [comment]
@dylanwh Sure! I will keep finding something to contribute to 😁
Thanks a lot!

**Comment by dylanwh** [comment]
I just filed [Bug 1427576](https://bugzilla.mozilla.org/show_bug.cgi?id=1427576), which is meant to be open-ended: reviewing our error messages and making them better match the style guide for Firefox. Let me know on that bug if it sounds interesting.

In particular some of our error messages are quiet frightening.

**Comment by dylanwh** [comment]
By "open ended" I mean we'd accept small PRs for individual messages. I don't expect a single person to review *all* of them.

**Comment by Fatehsandhu** [comment]
@dylanwh Sure I will have a look at them in the morning.

**Pull request merged**